src/sidebar/services/groups.js

```diff
@@ -106,14 +107,20 @@ export default function groups(
         directLinkedGroupId = null;
       }
     }
-
     // If service groups are specified only return those.
     // If a service group doesn't exist in the list of groups don't return it.
     if (svc && svc.groups) {
-      const focusedGroups = groups.filter(
-        g => svc.groups.includes(g.id) || svc.groups.includes(g.groupid)
-      );
-      return focusedGroups;
+      try {
+        // The groups may not be available if they are being fetched via RPC.
+        // See mutateAsyncGroups in fetch-config.js
+        const focusedGroups = await svc.groups;
+        const filteredGroups = groups.filter(
+          g => focusedGroups.includes(g.id) || focusedGroups.includes(g.groupid)
+        );
+        return filteredGroups;
+      } catch (e) {
+        toastMessenger.error(e.message);
```

Comment:
Since we're not returning anything here, which groups end up getting shown in the LMS context? Presumably all the groups for the different courses that the user is a member of?

Reply:
check out extra-config-params branch in via3, then it should work, I'll try it out too and see if we should do something like filter to nothing perhaps?

Reply:
Oh also, need to force the group value to `$rpc:requestGroups` or just comment out that condition to see the RPC request work.